tests/integration/tests/kiali_metrics_test.go

```diff
@@ -0,0 +1,74 @@
+package tests
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/kiali/kiali/tests/integration/utils"
+)
+
+var METRICS_PARAMS = map[string]string{"direction": "outbound", "reporter": "destination"}
+
+func TestNamespaceMetrics(t *testing.T) {
+	assert := assert.New(t)
+
+	pollErr := wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		metrics, err := utils.NamespaceMetrics(utils.BOOKINFO, METRICS_PARAMS)
+		assert.Nil(err)
+		assert.NotEmpty(metrics)
+		return CheckMetrics(metrics.ResponseSize, metrics.RequestSize), nil
```

Comment:
Do we expect metrics to be empty initially and then to keep polling until they are non-empty? Similar to the recent flake fix, instead of an assertion here, it'd be better to return false if metrics are empty or err if `err != nil`. Otherwise the tests will fail from the assertion in the polling loop if the metrics are ever empty even if they are eventually non-empty.

Reply:
Here, even if metrics do not exist, the 'error' and 'metrics' objects should not be nil, only the particular metrics as attributes of 'metrics' object are empty.

Reply:
Ah got it, thanks. I don't think `assert` will fail the test. iirc `require` does that. You might want to return an `err` anyways to exit the polling or else use `require`. Otherwise the `metrics.ReponseSize` and `metrics.RequestSize` on line 23 will likely panic.

Reply:
removed those assertions, returning err